kratos/mpi/tests/sources/test_pointer_communicator.cpp

```diff
@@ -186,9 +187,67 @@ KRATOS_TEST_CASE_IN_SUITE(PointerCommunicatorConstructByFunctor, KratosMPICoreFa
     }
 
 
-};
+}
 
+KRATOS_TEST_CASE_IN_SUITE(PointerMapCommunicator, KratosMPICoreFastSuite)
+{
+    DataCommunicator& r_default_comm = ParallelEnvironment::GetDefaultDataCommunicator();
+    Model current_model;
+    auto& mp = current_model.CreateModelPart("mp");
+    mp.AddNodalSolutionStepVariable(PARTITION_INDEX);
+
+    const int world_size = r_default_comm.Size();
+    const int current_rank = r_default_comm.Rank();
 
+    auto pnode = mp.CreateNewNode(current_rank+1, current_rank,current_rank,current_rank); //the node is equal to the current rank;
+    pnode->FastGetSolutionStepValue(PARTITION_INDEX) = current_rank;
+    pnode->SetValue(TEMPERATURE, current_rank);
+
+    //we will gather on every node the global pointers of the nodes with index from
+    //current_rank(+1) to world_size
+    std::vector<int> indices;
+    for(int i=1; i<=world_size; ++i) {
+        indices.push_back(i);
+    }
+
+    auto gp_list = GlobalPointerUtilities::RetrieveGlobalIndexedPointers(mp.Nodes(), indices, r_default_comm);
+
+    // creates the gp map
+    GlobalPointerMapCommunicator<Node<3>, double> pointer_map_comm(r_default_comm, gp_list, 0.0);
+
+    // creates apply proxy
+    auto apply_temperature_assemble_proxy = pointer_map_comm.GetApplyProxy(
+        // local gp value update method
+        [](Node<3>& rNode, const double& NewValue) {
+            rNode.GetValue(TEMPERATURE) += NewValue;
+        },
+        // non local gp map update lambda method
+        [](double& rCurrentValue, const double NewValue) {
+            rCurrentValue += NewValue;
+        });
+
+    GlobalPointersUnorderedMap<Node<3>, double> assembly_values_map;
+    for (const auto& r_gp : gp_list.GetContainer()) {
+        assembly_values_map[r_gp] = (r_gp.GetRank() + 1.0) * 2;
+    }
+
+    // updates local gps and stores remote gps in a map for future communication
+    apply_temperature_assemble_proxy.Update(assembly_values_map);
```

Comment:
I don't love the name "Update". Maybe SetMap or SetMapValue?

I understand that in any case those functions cannot change the "struture" of the gp_map, that is, to add or take out entries. That's correct?
If that is not so, we will need to have a function "finalize"

Reply:
> I understand that in any case those functions cannot change the "struture" of the gp_map, that is, to add or take out entries. That's correct?

Yes that is correct, it is not allowed to take any entries to be added or removed.

Reply:
I still don't love the name ;-) ... anything better than update?

Reply:
Assign maybe?

Reply:
done as discussed :)